### PR TITLE
refactor `ovirtapi` to separate concerns

### PR DESCRIPTION
**Pull request by sjd78**
`ovirtapi` has been split to have transport, data transform and API interactions in separate files:

 - HTTP transport functions are in `ovirtapi/transport.js`
 - data transform functions (from API to internal structures and back again) are in `ovirtapi/transform.js`
   - convertInt() in API transform for some values that
     transport as strings
 - API interaction functions are in `ovirtapi/index.js`
 - `ovirtapi/index.js` presents the save facade as the previous `ovirtapi.js`
 - create `ovirtapi` types, initially applied to the transform
   functions
   - each transform object has 2 corresponding types:
      - Api_Type to define the type in and out of the REST API
      - _Type to define the type within the application/redux
   - Most of the types are stubbed to `Object` so the names can be used and the actual contents can be defined from the REST API definitions and transform functions at a later time.

### CODE REVIEW

File: src/ovirtapi/transform.js [hunk 1]
```diff
@@ -0,0 +1,615 @@
...
+        id: cdrom.file.id,
+      },
+    }
+  },
+}
+
+//
+// (only for a Storage Domain type === 'iso')
+const StorageDomainFile = {
+  toInternal ({ file }: { file: Object }): Object {
```
Comment (bond95) [inline src/ovirtapi/transform.js]: `InternalStorageDomainFileType`

File: src/ovirtapi/transform.js [hunk 2]
```diff
@@ -0,0 +1,615 @@
...
+    }
+  },
+
+  toApi: undefined,
+}
+
+//
+//
+const Cluster = {
+  toInternal ({ cluster }: { cluster: Object }): Object {
```
Comment (bond95) [inline src/ovirtapi/transform.js]: `InternalClusterType`

File: src/ovirtapi/transform.js [hunk 3]
```diff
@@ -0,0 +1,615 @@
...
+    }
+  },
+
+  toApi: undefined,
+}
+
+//
+//
+const Nic = {
+  toInternal ({ nic }: { nic: Object }): Object {
```
Comment (bond95) [inline src/ovirtapi/transform.js]: `InternalNicType`

File: src/ovirtapi/transform.js [hunk 4]
```diff
@@ -0,0 +1,615 @@
...
+    return {
+      id: nic.id,
+      name: nic.name,
+      vnicProfile: {
+        id: nic.vnic_profile ? nic.vnic_profile.id : null,
+      },
+    }
+  },
+
+  toApi ({ nic }: { nic: Object }): Object {
```
Comment (bond95) [inline src/ovirtapi/transform.js]: `ApiNicType`

File: src/ovirtapi/transform.js [hunk 5]
```diff
@@ -0,0 +1,615 @@
...
+      }
+    }
+    return res
+  },
+}
+
+//
+//
+const VNicProfile = {
+  toInternal ({ vnicProfile }: { vnicProfile: Object }): Object {
```
Comment (bond95) [inline src/ovirtapi/transform.js]: `InternalVNicProfileType`

File: src/ovirtapi/transform.js [hunk 6]
```diff
@@ -0,0 +1,615 @@
...
+    return vnicProfileInternal
+  },
+
+  toApi: undefined,
+}
+
+//
+//
+const Network = {
+  toInternal ({ network }: { network: Object }): Object {
```
Comment (bond95) [inline src/ovirtapi/transform.js]: `InternalNetworkType`

File: src/ovirtapi/transform.js [hunk 7]
```diff
@@ -0,0 +1,615 @@
...
+    }
+  },
+
+  toApi: undefined,
+}
+
+//
+//
+const Host = {
+  toInternal ({ host }: { host: Object }): Object {
```
Comment (bond95) [inline src/ovirtapi/transform.js]: Same.

File: src/ovirtapi/transform.js [hunk 8]
```diff
@@ -0,0 +1,615 @@
...
+    }
+  },
+
+  toApi: undefined,
+}
+
+//
+//
+const OS = {
+  toInternal ({ os }: { os: Object }): Object {
```
Comment (bond95) [inline src/ovirtapi/transform.js]: Same

File: src/ovirtapi/transform.js [hunk 9]
```diff
@@ -0,0 +1,615 @@
...
+    }
+  },
+
+  toApi: undefined,
+}
+
+//
+//
+const Icon = {
+  toInternal ({ icon }: { icon: Object }): Object {
```
Comment (bond95) [inline src/ovirtapi/transform.js]: Same.

File: src/ovirtapi/transform.js [hunk 10]
```diff
@@ -0,0 +1,615 @@
...
+    }
+  },
+
+  toApi: undefined,
+}
+
+//
+//
+const SSHKey = {
+  toInternal ({ sshKey }: { sshKey: Object }): Object {
```
Comment (bond95) [inline src/ovirtapi/transform.js]: Same.

File: src/ovirtapi/transform.js [hunk 11]
```diff
@@ -0,0 +1,615 @@
...
+    }
+  },
+
+  toApi: undefined,
+}
+
+//
+//
+const VmConsoles = {
+  toInternal ({ consoles }: { consoles: Object }): Array<Object> {
```
Comment (bond95) [inline src/ovirtapi/transform.js]: Same.

File: src/ovirtapi/transform.js [hunk 12]
```diff
@@ -0,0 +1,615 @@
...
+    }).sort((a: Object, b: Object): number => b.protocol.length - a.protocol.length) // Hack: 'VNC' is shorter then 'SPICE'
+  },
+
+  toApi: undefined,
+}
+
+//
+//
+const VmSessions = {
+  toInternal ({ sessions }: { sessions: Object }): Object {
```
Comment (bond95) [inline src/ovirtapi/transform.js]: Same.

File: src/ovirtapi/transform.js [hunk 13]
```diff
@@ -0,0 +1,615 @@
...
+
+//
+//
+type CloudInitInternalType = {
+  enabled: boolean,
+  hostName: string,
+  sshAuthorizedKeys: string
+}
+const CloudInit = {
+  toInternal ({ vm }: { vm: Object }): CloudInitInternalType {
```
Comment (bond95) [inline src/ovirtapi/transform.js]: Maybe `InternalCloudInitType`?

File: src/ovirtapi/transform.js [hunk 14]
```diff
@@ -0,0 +1,615 @@
...
+  if (val) {
+    return typeof val === 'number' ? val : Number.parseInt(val, 10)
+  }
+  return defaultValue
+}
+
+//
+//
+type ApiVmType = Object
+type InternalVmType = Object
```
[the remark this answers is not in the document]
Comment (sjd78) [reply src/ovirtapi/transform.js]: I totally agree!  I just didn't want to take the time to do that yet.  

What I can do now is put the stub types in place in another file and leave most of the as `Object` for now.  That way, the types can be filled out later as time allows.

[the remark this answers is not in the document]
Comment (bond95) [reply src/ovirtapi/transform.js]: You can actually write full type for VM model as the most important, and others in next time.

**Comment by sjd78** [comment]
@bond95 - I updated the PR so all of the transform functions have type names coming from __src/ovirtapi/types.js__.  I didn't add specific data type definitions beyond `Object` for most of the types.

Also, I dropped the "Internal" prefix from type names.  So "internal" types have no prefix while the REST/API types have "Api" prefixes.  I think it looks better that way.

**Comment by gregsheremeta** [comment]
Don't merge yet. I need to release 1.4.1 today first.

**Comment by gregsheremeta** [comment]
new commits since approval, so needs re-review. @bond95

**Comment by sjd78** [comment]
Rebased and squashed.  No code change since @bond95's review.

**Comment by sjd78** [comment]
@gregsheremeta - This PR is reviewed and ready to merge. There are no overlaps with #654 or #660.

**Pull request merged**